### ISSUE TITLE
fix(cyclomatic/elixir): catch-all arms inflate CCN

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#1272]
## Summary

Elixir's catch-all arm counts toward standard cyclomatic complexity, while every sibling language excludes its equivalent default arm. On `main` (d86d3806):

```elixir
case x do
  1 -> :a
  2 -> :b
  _ -> :c
end
```

reports cyclomatic sum **4** (base 1 + 3 `stab_clause`s). The byte-equivalent Rust `match` (`_ =>` filtered), Python `match` (`case _:` filtered), Ruby `case/in` (`in _` filtered, #977), Kotlin `when` (`else ->` filtered), C# switch expression (`_ =>` filtered, #282), Bash `case` (`*)` filtered, #211), and C-family `default:` all report **3**.

`cond` has the same shape with its idiomatic `true -> …` final arm:

```elixir
cond do
  x > 10 -> :big
  x > 5 -> :mid
  true -> :small
end
```

also reports 4 where an `if/elif/else` sibling reports 3.

## Root cause

`src/metrics/cyclomatic/elixir.rs` counts every `E::StabClause` (except the anonymous-fn head clause, #776) with no bare-wildcard filter — the lesson-11 / parity-family-5 exclusion that every other switch-bearing language carries is missing.

## Fix plan

1. Add a helper (next to `elixir_is_anonymous_fn_head_clause`) that classifies a `stab_clause` as a bare catch-all: its left-hand side is a single `_` pattern (underscore identifier) with no `when` guard. A guarded `_ when g ->` still counts (matches Rust's `_ if g` rule); `{_, x} ->` or `_x ->` binding patterns still count.
2. Gate the `E::StabClause` standard-CCN arm on it.
3. Decide and document the `cond` `true ->` arm: it is the construct's designated default (mirrors `default:`) but is also a literal truth test. Recommendation: exclude a literal `true ->` arm *only under a `cond` container* (the container is identifiable from the ancestors chain via `elixir_call_keyword == Some("cond")`); leave `true ->` under `case` counting, since there it is an ordinary pattern. If the narrower fix is chosen instead, say so in the arm comment.
4. Note the cognitive impl is already correct (container-only, arms free) — no change there.
5. Regression tests: `case` with `_ ->` at cyclo 3; `case` with `_ when g ->` at 4; multi-clause anonymous fn ending in `_ ->` (the head-clause skip and the new filter compose); `cond` with `true ->` per the decision in step 3. Test-via-revert.

---

## Resolution Plan

1. In `src/metrics/cyclomatic/elixir.rs`, add `elixir_is_bare_catchall_clause` next to `elixir_is_anonymous_fn_head_clause`: true when the `stab_clause`'s `left` field (`arguments`) holds exactly one pattern that is an `identifier` whose text is `_`, with no `when` guard node. Verify the exact LHS shape (`arguments` wrapper vs bare identifier, and where the `when` operator sits) with `bca dump` on `case`/`cond`/multi-clause-`fn` fixtures before writing the matcher — mirror `rust.rs`'s pattern-field check, not a text scan of the whole clause.
2. Gate the standard-CCN arm: `E::StabClause if elixir_is_anonymous_fn_head_clause(..) || elixir_is_bare_catchall_clause(..) => {}` (keep the two guards separate or short-circuited in that order; the head clause of `fn _ -> x end` must stay excluded for the head-clause reason, not the wildcard reason, so the comment stays truthful).
3. `cond` `true ->`: exclude a literal `true ->` arm only when the clause's container Call is `cond` — walk `ancestors` from the `stab_clause` through its `do_block` to the `Call` and test `elixir_call_keyword(call, code) == Some("cond")` (helper already exists at `src/metrics/cognitive.rs:583` and is imported by checker/getter siblings). `true ->` under `case` keeps counting (ordinary pattern there). Document the rule and the parity rationale (`default:` / `else` family, lesson 11 / grammar-dispatch rule 8) in the arm comment.
4. No cognitive change — `src/metrics/cognitive.rs`'s Elixir handling is container-based and already correct; state this in the PR description so reviewers don't hunt for a missing sibling edit.
5. Regression tests in the cyclomatic test module: (a) `case` with `1 / 2 / _` arms → sum 3 (currently 4); (b) `case` with `_ when g ->` → 4; (c) `case` with `_x ->` binding → 4; (d) multi-clause anonymous fn ending in `_ ->` — head-skip and wildcard filter compose (assert exact value); (e) `cond` ending `true ->` → 3; (f) `case` containing a `true ->` arm still counts it. Assert `cyclomatic_modified` unchanged throughout (container-only). Verify (a) and (e) by test-via-revert per `.claude/rules/testing.md`.
6. Run the full workspace suite (`cargo test --workspace --all-features`) — Elixir has no integration-corpus coverage, so no submodule snapshot churn is expected; confirm, then `make pre-commit`.

## Assessment

| Dimension  | Rating |
|------------|--------|
| Difficulty | Medium |
| Complexity | Low |
| Priority   | High |

---

## Resolution

Fixed in commit ffbf592b on `fix/issue-1272` (integration branch `fix/batch-2026-08-09`). The `E::StabClause` standard-CCN arm now skips (a) a sole unguarded `_ ->` catch-all in any container and (b) `cond`'s literal `true ->` final arm, anchored to the `cond` container via `elixir_call_keyword` over the `Ancestors` chain. Guarded `_ when g ->`, named discards `_x ->`, and `true ->` under `case` all still count. Both issue fixtures report 3, matching every sibling. Cognitive and modified CCN untouched. Elixir added to cyclomatic parity families 1/2/3/5; seven new unit tests plus three re-derived expectations, each verified failing under targeted perturbations.

**Comment by dekobon** [comment]
Fix plan (detail in body): add a bare-catch-all classifier for `stab_clause` (single `_` pattern, no `when` guard) beside `elixir_is_anonymous_fn_head_clause` and gate the standard-CCN arm on it; decide the `cond`/`true ->` question explicitly (recommendation in body: exclude only under a `cond` container). Cognitive needs no change. Medium-small; the guarded-discard and anonymous-fn-composition tests are the load-bearing part.

**Comment by dekobon** [comment]
Added resolution plan and assessment ratings (Difficulty: Medium, Complexity: Low, Priority: High). Verified live: the E::StabClause standard-CCN arm in src/metrics/cyclomatic/elixir.rs has the #776 head-clause skip but no bare-wildcard filter, while rust.rs/python.rs/ruby.rs siblings in the same directory all carry the exclusion. The cond true-> exclusion can reuse elixir_call_keyword (src/metrics/cognitive.rs:583) via the Ancestors chain the trait already receives. No labels applied — priority is High (systematic +1 CCN inflation on idiomatic Elixir case/cond).

**Comment by dekobon** [comment]
Fixed on branch `fix/issue-1272` (commit ffbf592b), pending merge to the `fix/batch-2026-08-09` integration branch.

**What changed** (`src/metrics/cyclomatic/elixir.rs`): the `E::StabClause` standard-CCN arm is now gated on two new classifiers beside the #776 head-clause skip:

- `elixir_is_bare_catchall_clause` — sole unguarded `_` pattern (`_ ->`), excluded in any container. A guarded `_ when g ->` still counts (the guard makes `left` a `binary_operator`, so the classifier rejects it structurally before any text check), and a named discard `_x ->` still counts (Rust's bare-`_`-only precedent).
- `elixir_is_cond_default_clause` — sole unguarded literal `true` pattern whose parent is the `do_block` of a `Call` spelling `cond` (reusing cognitive's `elixir_call_keyword` via the `Ancestors` chain, as the triage comment predicted — that path still holds). `true ->` under `case` keeps counting.

Both issue fixtures now report **3** (were 4), matching the sibling table exactly. Modified CCN is untouched (container-only), and cognitive needed no change (verified: container-based, arms already free).

**Decisions made**
- `_ ->` is excluded uniformly across containers (`case`, `receive`, multi-clause `fn` dispatch, `rescue`) — it is each construct's designated default. The `rescue _ ->` corner deliberately diverges from C-family `catch (...)` (documented in the arm comment; the form is vanishingly rare and `try` still pays modified).
- `cond`'s exclusion took the recommended container-anchored form (grammar-dispatch §8).
- Composition: in `fn 0 -> :zero; _ -> :other end` the head clause is skipped for the head reason and the `_ ->` for the wildcard reason (sum 4); a 3-clause `fn ... _ ->` counts exactly its middle clause (sum 5).

**Tests** (all watched failing under targeted perturbations)
- 6 new + 1 follow-up unit tests in `src/metrics/cyclomatic.rs`: guarded wildcard, named discard, `true ->` under `case`, `cond` without default, fn/catch-all composition, nested `case`-in-`cond` scoping, non-head `true ->` under `fn`; 3 existing tests re-derived (6→5, 6→5, 5→4).
- Elixir joined the lesson-11 parity families 1/2/3/5 in `tests/parity/cyclomatic_cross_language_parity.rs` with the Java-style `defmodule` class offset — the cross-language pin the issue's table was measured against.

Full workspace suite green (5297+ tests, no integration-snapshot churn as predicted); `BCA_GATE: pass (gate=pre-commit)`.
